### PR TITLE
tsp, refactor, move multipart in code-model to serializationFormats

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Moved from `usage` to `serializationFormats`.

```yaml
      usage:
        - input
        - public
      serializationFormats:
        - multipart
```

When we supports `xml`, it would also be in `serializationFormats`. (`xml` would have further data in `serialization` property)

This would be compatible with code-model from m4.

---

One existing problem here (same for m4) is that `json` is not in `serializationFormats`.
So the assumption is kind of -- if it is not specified otherwise (`xml` or `multipart`), it is `json`

prepare for https://github.com/Azure/autorest.java/issues/2473

### CODE REVIEW

File: typespec-extension/src/main/java/com/azure/autorest/TypeSpecPlugin.java [hunk 1]
```diff
@@ -159,7 +158,7 @@ protected void writeHelperClasses(Client client, JavaPackage javaPackage, JavaSe
         // MultipartFormDataHelper
         final boolean generateMultipartFormDataHelper = client.getModels().stream()
                 .filter(ModelUtil::isGeneratingModel)
-                .anyMatch(m -> m.getImplementationDetails() != null && m.getImplementationDetails().getUsages().contains(ImplementationDetails.Usage.MULTIPART_FORM_DATA));
+                .anyMatch(m -> m.getSerializationFormats().contains("multipart"));
```
Comment (haolingdong-msft) [inline typespec-extension/src/main/java/com/azure/autorest/TypeSpecPlugin.java:161]: "multipart" -> use constant?

Comment (weidongxu-microsoft) [reply typespec-extension/src/main/java/com/azure/autorest/TypeSpecPlugin.java]: fixed via `KnownMediaType.MULTIPART.value()`

**Review by haolingdong-msft — APPROVED** [review]

File: javagen/src/main/java/com/azure/autorest/model/clientmodel/ClientModel.java [hunk 2]
```diff
@@ -670,6 +683,17 @@ public Builder crossLanguageDefinitionId(String crossLanguageDefinitionId) {
             return this;
         }
 
+        /**
+         * Sets the cross language definition id for the model.
```
Comment (XiaofeiCao) [inline javagen/src/main/java/com/azure/autorest/model/clientmodel/ClientModel.java:687]: nit, comment is wrong

Comment (weidongxu-microsoft) [reply javagen/src/main/java/com/azure/autorest/model/clientmodel/ClientModel.java]: Thanks. Fixed.

**Pull request merged**